Standards/scs-0402-v1-status-page-openapi-spec-decision.md

```diff
@@ -0,0 +1,107 @@
+
+## Decision
+
+### Common values
+
+Some defined schemas are used as common values and data types. These common values help to simplify the objects using these values by providing meaningful names, and centralize the data types of these fields. A change of ID type for example only needs one change in the common value, and not in any of the objects which include an ID.
+
+Special mentions:
+
+#### Id
+
+IDs are used for identification of resources, which can be retrieved by the API.
+
+UUIDs are used, to ensure uniqueness and they can be visualy recognized as identifier.
+
+#### IdList
+
+List of IDs are used to reference other resources in an API object.
+
+#### Incremental
+
+The `Incremental` value is used in combination with an ID to identify a sub resource of any kind, or define a given order. `Incremental`s are not globaly unique, but unique for every sub resource of an unique resource.
+
+### API objects
+
+All objects which are used as payload, either as request or response, are defined by schemas. This centralizes the maintanence of field names and types, for both requests and responses.
+
+The only exception are responses which return lists. The list of API objects is only used once and the definition of an array type is trivial.
+
+### API object fields
+
+Most fields of objects are not required. This allows usage as request and response payloads.
+
+Responses of payload objects, which contain an ID or an `Incremental` typed field, MUST fill the ID or `Incremental` field to fully identify the (sub) resource.
+
+Requests on a single resource MUST contain the ID in the path parameters. Request on sub resources MUST contain the ID and `Incremental` typed value as path parameters. The payload SHOULD NOT contain the ID or `Incremental` typed field.
+If it contains these fields as payload, they SHALL NOT change them.
+
+Requests to updating operations SHOULD contain the minimum of the changed fields, but MAY contain the full object. ID and `Incremental` typed fields MUST follow the same rules as stated above.
+
+### Endpoint naming
+
+The endpoints are named in plural form, even when handeling single objects, to keep uniform paths.
+
+### Phase list
+
+The list of phases an incident can traverse has a crucial order. So it MUST be handled as the given list.
+
+Delete or update operations are FORBIDDEN.
+
+To "change" a phase list, a new one must be created. The old one must be kept. For this mechanic the lists are structured in generations. All references to phases MUST include their generation to ensure correct references.
+
+### Labels
+
+Labes are identifying meta data to components. They do not represent a resource or sub resource of any kind. They are designed as non system critical pieces of information, mainly intented for humans.
+
+They represent a key value pair as structured data for a single component. Same value keys and values do not reference each other necessarily.
```

Comment:
> Same value keys and values do not reference each other necessarily.

Is this meant to say that no database references are done, here?

Reply:
Exactly :+1:

Reply:
Let's rephrase this bit, then. Like:

```suggestion
Labels are simple key/value pairs attached to components, categorizing them dynamically.
```

(I really like the definition of K8s labels, btw: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/)